src/op_handlers/far_call.rs

```diff
@@ -102,10 +102,10 @@ fn far_call_params_from_register(
     let mut ergs_passed = source.0[3] as u32;
     let gas_left = vm.gas_left()?;
 
-    if ergs_passed > gas_left {
-        ergs_passed = (gas_left * FAR_CALL_GAS_SCALAR_MODIFIER_DIVIDEND)
-            / FAR_CALL_GAS_SCALAR_MODIFIER_DIVISOR;
-    }
+    let maximum_gas =
+        gas_left / FAR_CALL_GAS_SCALAR_MODIFIER_DIVISOR * FAR_CALL_GAS_SCALAR_MODIFIER_DIVIDEND;
```

Comment:
Careful here. If `gas_left < 64` this will be `0`, while it should be in almost all cases non-`0`. For example, `63 * 63/64 == 62.015625`, which truncates to `62`, would result in `0` if you perform the integer division first.

Reply:
I see that could be an issue. The thing is, both implementations (vm1, vm2) do it this way.

Reply:
[See](https://github.com/matter-labs/vm2/blob/master/src/instruction_handlers/far_call.rs#L97-#L100)